CMakeLists.txt

```diff
@@ -44,54 +41,13 @@ if( POLICY CMP0054 )
   cmake_policy(SET CMP0054 OLD)
 endif()
 
-##############################################
-# Compiler support for <filesystem> in C++17:
-# https://en.cppreference.com/w/cpp/compiler_support
-#    GCC 7.1 has <experimental/filesystem>, link with -libc++experimental or -lstdc++fs
-#    GCC 8 has <filesystem>   link with -lstdc++fs
-#    GCC 9   expected to support <filesystem>
-#    Clang 4 (XCode10) has no support for <filesystem>, partial C++17
-#    Clang 7 has complete <filesystem> support for C++17
-#    Visual Studio 2017 15.7 (v19.14)supports <filesystem> with C++17
-#    MinGW has no support for filesystem.
-#
-# If we have support for <filesystem> and C++17, turn on the C++17 standard flag, 
-# else set standard to C++11 and install boost filesystem
-# Also allow external library for <filesystem>
-
-set(extra_lib_for_filesystem)   # sometimes -libc++experimental or -lstdc++fs
-set(INTERNAL_CPP_STANDARD "c++11")
-set(NEEDS_BOOST ON)
-
-### comment this out and save for next PR
-#if(${CMAKE_CXX_COMPILER_ID} STREQUAL "GNU")
-#    if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL "9")
-#         set(INTERNAL_CPP_STANDARD "c++17")
-#	 set(NEEDS_BOOST OFF)
-#    elseif(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL "8")
-#         set(INTERNAL_CPP_STANDARD "c++17")
-#	 set(extra_lib_for_filesystem "-lstdc++fs")
-#	 set(NEEDS_BOOST "OFF")
-#    endif()	 
-#elseif(${CMAKE_CXX_COMPILER_ID} MATCHES "Clang")
-#    if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL "7")
-#         set(INTERNAL_CPP_STANDARD "c++17")
-#	 set(NEEDS_BOOST OFF)
-#    endif()
-#elseif(${CMAKE_CXX_COMPILER_ID} STREQUAL "MSVC")
-#      if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL "19.14")
-#            set(INTERNAL_CPP_STANDARD "c++17")
-#	    set(NEEDS_BOOST OFF)
-#      endif()
-#endif()
-
-
 ##############################################
 # Global NuPIC CMake options
 #
-option(NUPIC_BUILD_PYEXT_MODULES
-       "Turn on building of python extension modules for nupic.bindings; turn off to build only static nupic_core lib with full symbol visibility."
-       ON)
+option(USE_CPP11 "Force compiler to use C++11 standard." ON)
+option(NEEDS_BOOST "Force compiler to install and use Boost." ON)
+option(PYTHON2_BUILD "Toggle whether to build for Python2.7, default ON." ON)
+option(PYTHON3_BUILD "Toggle whether to build for Python3.x, default ON." ON)
```

Comment:
My thinking is that the builder may want to dictate these options but provide defaults.
- USE_CPP11  Allows me to force C++11 while testing...otherwise I would always build with C++17.  Normally this would be OFF and it would use C++17 if available.
- NEEDS_BOOST Allows the user to force including boost in the nupic_core library so that his app can use some of the features.  Also allows me to force testing...otherwise it would never use boost on MSVC.  Normally this would default to OFF and it would be turned ON only if their compiler did not support std::filesystem.
- PYTHONx_BUILD Allows me to test building one or the other or both Python interface libraries.  A user may want to not build either if this is to be used for a C++ only application. Normally these would both be ON.

Reply:
It makes sense to be able to force specific behavior, be it for testing or user wants it. Thanks for explanation.